staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1428,35 +1436,120 @@ func (c *cacheWatcher) process(ctx context.Context, resourceVersion uint64) {
 	}
 }
 
+type status int
```

Comment:
I'm somewhat reluctant with this PR.

It seems that you manually resolved these conflicts. Instead, I would like to ensure that if we would also cherrypick https://github.com/kubernetes/kubernetes/pull/108414 this would patch clearly...

Reply:
Makes sense, let me close the 1.23 and 1.24 cherry-picks and open new ones with both 108414 and 116024.